src/book/02-projects/01-nubots/12-mathematics.mdx

```diff
@@ -0,0 +1,128 @@
+u_3^a \end{bmatrix}
+$$
+
+To make computations such as inner product or cross product with coordinates, they must be relative to the same basis.
+
+## Rotations
+
+To describe the motion of the robot, we need a **reference frame** and a **coordinate system**.
+
+The reference frame is the physical rigid body and the coordinate system is a mathematical concept, which is represented as a basis.
+
+**Rotations** are defined by a 3x3 matrix. If we have a vector $\mathbf{u}^a$, that is the vector $u$ in the coordinate system $a$, the rotation
+
+$$
+\begin{bmatrix}
+\cos(\psi) & -\sin(\psi) & 0 \\
+\sin(\psi) & \cos(\psi) & 0 \\
+0 & 0 & 1 \end{bmatrix}
+$$
+
+will rotate $\mathbf{u}^a$ by $\psi$ around the coordinate axis $\mathbf{a_3}$.
+
+The orientation of the basis $b$ relative to $a$ can be described by three consecutive rotations about the main axis. There are 12 different orders to do these rotations in, and each triplet of rotated angles is called a set of **Euler angles**. In robotics, the triplet is **yaw**, **pitch** and **roll** (Z-Y-X).
+
+In the NUbots codebase, we denote a rotation matrix from space $b$ to space $a$ as `Rab`.
+
+There rotation matrices belong to the **special orthogonal group of order 3**. These all have the following properties:
+
+$$
+\mathbf{R}^{-1} = \mathbf{R}^T
+$$
+
+$$
+\text{det}(\mathbf{R}) = 1
+$$
+
+If we have a vector $\mathbf{u}$ (which goes from point C to point D) in a space $a$ and we want to rotate it from space $a$ to space $b$, we write:
+
+$$
+\mathbf{u}^b = \mathbf{R}^b_a \cdot \mathbf{u}^a
+$$
+
+This is written in the NUbots codebase as `rCDb = Rba * rCDa`.
+
+## Transformations
+
+A transformation matrix in three dimensions is a 4x4 matrix containing a **rotation** component and a **translation** component.
+
+If we have some rotation $\mathbf{R}_a^b$ and a translation component $\mathbf{u}^b$, then we have a transformation matrix that transforms from space $a$ to space $b$:
+
+$$
+\mathbf{H}_a^b = \begin{bmatrix}
+\mathbf{R}_a^b & \mathbf{u}_{BA}^b \\
+\mathbf{0} & 1
+\end{bmatrix}
+$$
+
+This is written in the NUbots codebase as `Hba`.
+
+The vector $\mathbf{u}_{BA}^b$ is the vector from A (the origin of space a) to B (the origin of space b) in b space, which would be written in the codebase as `rBAb`.
+
+The reason for this is that if we apply the transform $H^b_a$ to some point $\rho^a$, then $\rho^a$ will first be rotated by $R_a^b$, giving $\rho^b$. $\rho^b$ will then be translated by some vector according to the transformation matrix. Since $\rho^b$ is now rotated in the $b$ coordinate system, the translation vector must also be in this $b$ coordinate system too. Thus, the translation vector is $\mathbf{u}_{BA}^b$.
+
+Transformation matrices left multiply with vectors to get a vector in the new space.
+
+$$
+\mathbf{v}^b = \mathbf{H}^b_a \cdot \mathbf{v}^a
+$$
+
+This is written in the codebase as `vCDb = Hba * vCDa`, if $\mathbf{v}$ represents a vector from point D to point C.
```

Comment:
We do in some places use `v` rather than `r`, usually when we are representing a velocity (see https://github.com/NUbots/NUbots/blob/master/module/platform/darwin/SensorFilter/src/MotionModel.h#L43 and https://github.com/NUbots/NUbots/blob/biddulph/eigenify_platform/module/platform/darwin/SensorFilter/src/MotionModel.h#L52 for a comparison, `rTWw` vs `vTw`).

I changed `rCDb = Hba * rCDa` to `vCDb = Hba * vCDa` because you have `\mathbf{v}^b` and `\mathbf{v}^a` in the equation above. But it might be better if we changed

`\mathbf{v}^b = \mathbf{H}^b_a \cdot \mathbf{v}^a` to `\mathbf{r}^{b}_{BA} = \mathbf{H}^b_a \cdot \mathbf{r}^{a}_{BA}`

Reply:
Yeah I think changing the name of the vector to `r` in the equations would work better then.